File: uniapp_flutter/lib/utils/media_action_type.dart
enum MediaActionType {
  DOWNLOAD,
  UPLOAD
}

class MediaActionTypeHelper {
  static int getValue(MediaActionType status){
    switch(status){
      case MediaActionType.DOWNLOAD:
        return 1;
      case MediaActionType.UPLOAD:
        return 2;
      default:
        return 0;
    }
  }
  static final Map<int, MediaActionType> valueToStateNameMap = new Map();
  final int value = 0;
  static createMap() {

    for (MediaActionType myEnum in MediaActionType.values) {
      valueToStateNameMap[MediaActionTypeHelper.getValue(myEnum)]= myEnum;
    }
  }

  static MediaActionType getStateByValue(int value) {
    if(valueToStateNameMap.isEmpty) {
      createMap();
    }
    return valueToStateNameMap[value];
  }
}

File: app/app/src/main/java/com/vassar/unifiedapp/utils/MediaActionType.java
package com.vassar.unifiedapp.utils;
import java.util.HashMap;
import java.util.Map;

public enum MediaActionType {

    UPLOAD(1),
    DOWNLOAD(0);

    private static final Map<Integer, MediaActionType> valueToActionNameMap = new HashMap<>();
    private final Integer value;

    static {
        for (MediaActionType myEnum : values()) {
            valueToActionNameMap.put(myEnum.getValue(), myEnum);
        }
    }

    private MediaActionType(final Integer newValue) {
        value = newValue;
    }

    public Integer getValue() {
        return value;
    }

    public static MediaActionType getActionByValue(Integer value) {
        return valueToActionNameMap.get(value);
    }
}
